lib/Element.js

```diff
@@ -279,13 +277,24 @@ Element.prototype.toReact = function (index) {
     }
   }))
 
-  return React.createElement(this.nodeName, props, this.text || this.children.map(function (el, i) {
-    if (el instanceof Element) {
-      return el.toReact(i)
-    } else {
+  var children
+
+  if (this.text) {
+    children = this.text
+  } else if (this.children.length === 1) {
+    var el = this.children[0]
+    children = el instanceof Element ? el.toReact() : el
+  } else if (this.children.length > 1) {
+    children = this.children.map(function (el) {
+      if (el instanceof Element) {
+        return el.toReact(el.key)
+      }
+
       return el
-    }
-  }))
+    })
+  }
+
+  return React.createElement(this.nodeName, props, children)
```

Comment:
This is where I would like to introduce my little refactor, simplifying and moving into it's own function: https://github.com/Olical/react-faux-dom/pull/74/files#diff-52cea43ae897a1705ec51162aed25f63R276